code/game/objects/items/devices/scanners/plant_analyzer.dm

```diff
@@ -0,0 +1,275 @@
+			"nutri" = tray.reagents.total_volume,
+			"nutri_max" = tray.maxnutri,
+			"yield_mod" = tray.yieldmod,
+			"weeds" = tray.weedlevel,
+			"weeds_max" = MAX_TRAY_WEEDS,
+			"pests" = tray.pestlevel,
+			"pests_max" = MAX_TRAY_PESTS,
+			"toxins" = tray.toxic,
+			"toxins_max" = MAX_TRAY_TOXINS,
+			"reagents" = list(),
+		)
+		for(var/datum/reagent/reagent in tray.reagents.reagent_list)
+			last_scan_data["tray_data"]["reagents"] += list(list(
+				"name" = reagent.name,
+				"volume" = reagent.volume
+			))
+
+	if(seed)
+		last_scan_data["seed_data"] = make_seed_data(seed)
+
+	if(graft)
+		last_scan_data["graft_data"] = list(
+			"name" = graft.parent_name,
+			"icon" = graft.icon,
+			"icon_state" = graft.icon_state,
+			"yield" = graft.yield,
+			"production" = graft.production,
+			"lifespan" = graft.lifespan,
+			"endurance" = graft.endurance,
+			"weed_rate" = graft.weed_rate,
+			"weed_chance" = graft.weed_chance,
+			"graft_gene" = graft.stored_trait.type
+		)
+
+	ui_interact(user)
+
+	return TRUE
+
+
+/obj/item/plant_analyzer/proc/make_seed_data(obj/item/seeds/seed)
+	var/list/seed_data = list(
+		"name" = seed.plantname,
+		"icon" = seed.growing_icon,
+		"icon_state" = seed.icon_harvest,
+		"product" = seed.product?.name,
+		"product_icon" = seed.product?.icon,
+		"product_icon_state" = seed.product?.icon_state,
+		"potency" = seed.potency,
+		"yield" = seed.yield,
+		"instability" = seed.instability,
+		"maturation" = seed.maturation,
+		"production" = seed.production,
+		"lifespan" = seed.lifespan,
+		"endurance" = seed.endurance,
+		"weed_rate" = seed.weed_rate,
+		"weed_chance" = seed.weed_chance,
+		"graft_gene" = seed.graft_gene ? seed.graft_gene : /datum/plant_gene/trait/repeated_harvest
+	)
+	seed_data["removable_traits"] = list()
+	seed_data["core_traits"] = list()
+	for(var/datum/plant_gene/trait/trait in seed.genes)
+		if(trait.mutability_flags & PLANT_GENE_REMOVABLE)
+			seed_data["removable_traits"] += trait.type
+		else
+			seed_data["core_traits"] += trait.type
+	seed_data["reagents"] = list()
+	for(var/datum/plant_gene/reagent/reagent in seed.genes)
+		seed_data["reagents"] += list(list(
+			"name" = reagent.name,
+			"rate" = reagent.rate
+		))
+	seed_data["volume_mod"] = (locate(/datum/plant_gene/trait/maxchem) in seed.genes) ? 2 : 1
+	seed_data["mutatelist"] = list()
+	for(var/obj/item/seeds/mutant as anything in seed.mutatelist)
+		seed_data["mutatelist"] += initial(mutant.plantname)
+	if(seed.product)
+		var/obj/item/food/grown/product = new seed.product
+		var/datum/reagent/product_distill_reagent = product.distill_reagent
+		seed_data["distill_reagent"] = initial(product_distill_reagent.name)
+		var/datum/reagent/product_juice_typepath = product.juice_typepath
+		seed_data["juice_name"] = initial(product_juice_typepath.name)
+		seed_data["grind_results"] = list()
+		for(var/datum/reagent/reagent as anything in product.grind_results)
+			seed_data["grind_results"] += initial(reagent.name)
+		qdel(product)
```

Comment:
We make a ghost item every time we scan a seed? This seems kind of wasteful, maybe it should be cached or something. 
```dm
var/static/list/product_grinder_results = list()
if(!product_grinder_results[seed.product])
	var/obj/item/temp_item = new()
	product_grinder_results[seed.product] = list()
	product_grinder_results[seed.product]["juicing"] = ...
	product_grinder_results[seed.product]["grind"] = ...
	product_grinder_results[seed.product]["distilling"] = ...
	qdel(temp_item)
seed_data["distill_reagent"]["juice"] = product_grinder_results[seed.product]["juicing"]
...
```

Reply:
I'm not sure how to do this better.